app/models/concerns/user_omniauth_concern.rb

```diff
@@ -0,0 +1,52 @@
+module UserOmniauthConcern
```

Comment:
Ideally I'll just include more concerns like this and it should automatically work with more providers. This allows me to keep the implementation details of each provider separate.

There's two levels of abstraction in this concern. One deals with the user. One deals with the social network. They should live separately.